src/main/java/net/fabricmc/loom/build/MixinRefmapHelper.java

```diff
@@ -67,11 +67,33 @@ public static boolean addRefmapName(Project project, Path outputPath) {
 						MixinExtension.getMixinInformationContainer(sourceSet)
 				);
 
+				String[] rootPaths = (String[]) sourceSet.getResources().getSrcDirs().stream()
+						.map(root -> {
+							String rootPath = root.getAbsolutePath();
+
+							if (rootPath.charAt(rootPath.length() - 1) != File.separatorChar) {
+								rootPath += File.separatorChar;
+							}
+
+							return rootPath;
+						})
+						.toArray();
```

Comment:
```suggestion
				String[] rootPaths = sourceSet.getResources().getSrcDirs().stream()
						.map(root -> {
							String rootPath = root.getAbsolutePath();

							if (rootPath.charAt(rootPath.length() - 1) != File.separatorChar) {
								rootPath += File.separatorChar;
							}

							return rootPath;
						})
						.toArray(String[]::new);
```
I'm not sure if the cast from `Object[]` is even safe, but this is nicer anyway

Reply:
I didn't know it could be done this way. At least I learned something today :)